internal/resources/azure/log_analytics_workspace.go

```diff
@@ -0,0 +1,236 @@
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+// BuildResource builds a schema.Resource from a valid LogAnalyticsWorkspace struct.
+// The returned schema.Resource can have 3 potential schema.CostComponent associated with it:
+//
+//		1. Log data ingestion, which can be either:
+//			a) Pay-as-you-go, which is only valid for a sku of PerGB2018 and uses a usage param
+//			b) Billed per commitment tiers, which is only valid for a sku of CapacityReservation
+//		2. Log retention, which is free up to 31 days and then is billed per GB of data retained after the free limit.
+//		   Additional GB used comes from the usage param.
+//		3. Data export, which is billed per monthly GB exported and is defined from a usage param.
+//
+// Outside the above rules - if the workspace has sku of Free we return as a free resource & if the workspace sku
+// is in a list of unsupported skus then we mark as skipped with a warning.
+func (r *LogAnalyticsWorkspace) BuildResource() *schema.Resource {
+	if r.SKU == skuFree {
+		return &schema.Resource{
+			Name:        r.Address,
+			IsSkipped:   true,
+			NoPrice:     true,
+			UsageSchema: LogAnalyticsWorkspaceUsageSchema,
+		}
+	}
+
+	if _, ok := unsupportedLegacySkus[strings.ToLower(r.SKU)]; ok {
+		log.Warnf("skipping %s as it uses legacy pricing options", r.Address)
+
+		return &schema.Resource{
+			Name:        r.Address,
+			IsSkipped:   true,
+			UsageSchema: LogAnalyticsWorkspaceUsageSchema,
+		}
+	}
+
+	var costComponents []*schema.CostComponent
+
+	if r.SKU == skuPerGB2018 {
+		costComponents = append(costComponents, r.logDataIngestion())
+	}
+
+	if r.SKU == skuCapacityReservation && r.ReservationCapacityInGBPerDay > 0 {
+		costComponents = append(costComponents, r.logDataIngestionFromCapacityReservation())
+	}
+
+	if r.RetentionInDays > logRetentionFreeTierLimit {
+		costComponents = append(costComponents, r.logDataRetention())
+	}
+
+	costComponents = append(costComponents, r.logDataExport())
+
+	return &schema.Resource{
+		Name:           r.Address,
+		UsageSchema:    LogAnalyticsWorkspaceUsageSchema,
+		CostComponents: costComponents,
+	}
+}
+
+func (r *LogAnalyticsWorkspace) logDataIngestionFromCapacityReservation() *schema.CostComponent {
+	return &schema.CostComponent{
+		Name:            "Log data ingestion",
+		Unit:            fmt.Sprintf("%d GB (per day)", r.ReservationCapacityInGBPerDay),
+		UnitMultiplier:  decimal.NewFromInt(1),
+		MonthlyQuantity: decimalPtr(decimal.NewFromFloat(30)),
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr(vendorName),
+			Region:        strPtr(r.Region),
+			Service:       strPtr(azureMonitorServiceName),
+			ProductFamily: strPtr(governanceProductFamily),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "skuName", Value: strPtr(fmt.Sprintf("%d GB Commitment Tier", r.ReservationCapacityInGBPerDay))},
+				{Key: "meterName", Value: strPtr(fmt.Sprintf("%d GB Commitment Tier", r.ReservationCapacityInGBPerDay))},
```

Comment:
TF docs mention `Must be in increments of 100 between 100 and 5000.`, so it's possible to set `1200` for example: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/log_analytics_workspace#reservation_capacity_in_gb_per_day. And the price won't be found.

This page https://docs.microsoft.com/en-gb/azure/azure-monitor/logs/manage-cost-storage#commitment-tiers mentions:
> Starting June 2, 2021, Capacity Reservations are now called Commitment Tiers. Data collected above your commitment tier level (overage) is now billed at the same price-per-GB as the current commitment tier level, lowering costs compared to the old method of billing at the Pay-As-You-Go rate, and reducing the need for users with large data volumes to fine-tune their commitment level. Three new commitment tiers were also added: 1000, 2000, and 5000 GB/day.

If I understand it correctly, it may mean that if we have 130 GB capacity, the first 100 GB should be billed for 100 GB tier, and 30 - for 200 GB.

Reply:
let me test if it's possible to set a `1200` value - It might be that the `azure` provider hasn't updated their docs to reflect the changes.

Reply:
yes it looks like this is possible to do so we'll have to convert the terraform value into a commitment tier. Your assumption about the tiering seems wrong though. From the pricing calculator 130 GB is billed entirely in the 100GB commitment tier:

![Screenshot 2022-02-14 at 13 10 48](https://user-images.githubusercontent.com/6455139/153870641-efa5278f-dc93-42b8-8ed7-7a24b81fd054.png)

which seems inline with what they say:

> data collected above your commitment tier level (overage) is now billed at the **same price-per-GB as the current commitment tier level**

I'll alter to reflect

Reply:
That's funny, 180GB is still 100, but 190 is 200 already :D
![image](https://user-images.githubusercontent.com/87082/153871351-c4c4379a-ae81-4ff2-ac37-7bc216776d5d.png)

Reply:
![](https://media.giphy.com/media/3o7527pa7qs9kCG78A/giphy.gif)

Reply:
added commitment tier translation: https://github.com/infracost/infracost/pull/1367/files#diff-a5a01adf10dff98c1abf28edb6847fbe6cd4ca5780bd938ec0e5bd4ec07cd33eR166-R185